extensions/eda/plugins/event_source/kafka.py

```diff
@@ -109,24 +109,49 @@ async def main(  # pylint: disable=R0914
     await kafka_consumer.start()
 
     try:
-        async for msg in kafka_consumer:
-            data = None
-            try:
-                value = msg.value.decode(encoding)
-                data = json.loads(value)
-            except json.decoder.JSONDecodeError:
-                data = value
-            except UnicodeError:
-                logger.exception("Unicode Error")
-
-            if data:
-                await queue.put({"body": data})
-            await asyncio.sleep(0)
+        await receive_msg(queue, kafka_consumer, encoding)
     finally:
         logger.info("Stopping kafka consumer")
         await kafka_consumer.stop()
 
 
+async def receive_msg(
+    queue: asyncio.Queue,
+    kafka_consumer: AIOKafkaConsumer,
+    encoding: str,
+) -> None:
+    """Receive messages from the Kafka topic and put them into the queue."""
+    logger = logging.getLogger()
+
+    async for msg in kafka_consumer:
+        event = {}
+
+        # Process headers
+        try:
+            headers = {header[0]: header[1].decode(encoding) for header in msg.headers}
+            event["headers"] = headers
```

Comment:
Can we place headers under meta like https://github.com/ansible/event-driven-ansible/blob/main/extensions/eda/plugins/event_source/webhook.py#L71?

Reply:
yes. that makes sense. When discussing metadata broadly, headers can be one of the elements included in the metadata. Thanks.